app/application.js

```diff
@@ -1,18 +1,53 @@
 import Mn from 'backbone.marionette';
 import Backbone from 'backbone';
-import tracksView from 'views/tracks';
+import Tracks from './collections/tracks';
+import Playlists from './collections/playlists';
+import AppLayout from './views/app_layout';
 
+require('./styles/app.styl');
 
-class Application extends Mn.Application {
+const Application = Mn.Application.extend({
 
-    onStart () {
-        console.log('start');
+    initialize: function () {
+        this.allTracks = new Tracks([], { type: 'all' });
+        this.upNext = new Tracks();
+        this.upNext.on('change:currentTrack', function() {
+            const upNext = app.upNext;
+            const currentTrack = upNext.getAttr('currentTrack');
+            if (!upNext.contains(currentTrack)) {
+                upNext.push(currentTrack);
+            }
+        });
+        this.allTracks.fetch({
+            success: function() {
+                app.allTracks.each(function(track) {
+                    app.upNext.add(track);
+                });
+            }
+        });
+
+        this.allPlaylists = new Playlists();
+        this.allPlaylists.fetch();
+
+        this.headerInfos = new Backbone.Model({
+            title: 'All Songs',
+            count: 0
+        });
+    },
+
+    onStart: function () {
         if (Backbone.history) {
             Backbone.history.start();
         }
-        const tracks = new tracksView();
-        tracks.render();
+        this.appLayout = new AppLayout();
+        this.appLayout.render();
+    },
+
+    switchPlaylist: function(collection) {
```

Comment:
Not sure you have to handle the playlist switch at top level. It's more a custom event that'll be triggered inside the underlying views and'll be catched by the dedicated playslit view.